src/nvme/ioctl.h

```diff
@@ -2917,8 +2918,9 @@ struct nvme_format_nvm_args {
 	enum nvme_cmd_format_pi pi;
 	enum nvme_cmd_format_pil pil;
 	enum nvme_cmd_format_ses ses;
-	__u8 lbaf;
-};
+	__u8 lbafu;
+	__u8 lbafl;
+} __attribute__((packed, aligned(__alignof__(__u32*))));
```

Comment:
No attributes are needed. We sorted the members according their natural size so we don't have any holes. But given
the argument structure size changes (our versioning scheme) we need also backwards compatibility, which seems to miss from this patch.

Reply:
In order to check the structure size (to determine version 1 or 2), I needed to keep the packed attribute.  Without it, the compiler would pad it out to the next 4 byte boundary.

Reply:
Fixed this but had to keep the packed attribute so the exact structure size could be used to differentiate between the version in use.

Reply:
Okay, I see the problem now.